src/app/hooks/useGlean.ts

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { useEffect } from "react";
+import Glean from "@mozilla/glean/web";
+import * as appEvents from "../../telemetry/generated/appEvents";
+
+// Custom hook that initializes Glean and returns the Glean objects required
+// to record data.
+export const useGlean = () => {
+  // Initialize Glean only on the first render
+  // of our custom hook.
+  useEffect(() => {
+    Glean.initialize("monitor-testing", true, {
+      // This will submit an events ping every time an event is recorded.
+      maxEvents: 1,
+    });
+
+    // Enable logging pings to the browser console.
+    Glean.setLogPings(true);
+
+    // Set the tag and sends pings to the debug ping viewer.
+    Glean.setDebugViewTag("monitor-testing");
+  }, []);
+
+  // Return all generated Glean objects required for recording data.
+  return {
+    appEvents,
+  };
```

Comment:
What's the benefit of wrapping it in another object - is this going to return additional data later?

```suggestion
  return appEvents;
```

Reply:
There could be depending on how this is set up. Since I am only exporting events from the same category it only creates one object. If you would add a different category header in the `metrics.yaml`, then it would generate two separate objects.

So yes for now we can return just the one object but over time this will likely become multiple objects with different categories.